hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/dispatcher/WritableStoreFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.dispatcher;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.service.consensus.ConsensusService;
+import com.hedera.node.app.service.consensus.impl.WritableTopicStore;
+import com.hedera.node.app.spi.state.WritableStates;
+import com.hedera.node.app.state.HederaState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+
+/**
+ * Factory for all stores. Besides providing helpers to create stores, this class also tracks, which
+ * {@link WritableStates} have been used.
+ */
+public class WritableStoreFactory {
```

Comment:
It is not clear to me what this class is adding. The docs say "this class also tracks, which {@link WritableStates}. have been used". But the HederaState does this also. What is it needed for here?

Reply:
Modified the javadoc. Since we will be having multiple writable stores, this class will have set of all those methods that call 
```
state.createWritableStates(Service.NAME);
```
This is similar to `ReadableStoreFactory`.